src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -590,9 +590,7 @@ private function validateClassNode($object, $cacheKey, ClassMetadataInterface $m
                     $object,
                     $cacheKey.':'.$propertyName,
                     $propertyMetadata,
-                    $propertyPath
-                        ? $propertyPath.'.'.$propertyName
-                        : $propertyName,
+                    PropertyPath::append($propertyPath, $propertyName),
```

Comment:
Right, I got confused with "UnitTest" - as this is not really a unit test. But yeah ok, we should be able to extend this one with coverage for that case.